pkg/descheduler/strategies/highnodeutilization.go

```diff
@@ -0,0 +1,157 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package strategies
+
+import (
+	"context"
+	"fmt"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/descheduler/pkg/api"
+	"sigs.k8s.io/descheduler/pkg/descheduler/evictions"
+	nodeutil "sigs.k8s.io/descheduler/pkg/descheduler/node"
+	"sigs.k8s.io/descheduler/pkg/utils"
+)
+
+// HighNodeUtilization evicts pods from under utilised nodes so that scheduler can schedule according to its strategy.
+// Note that CPU/Memory requests are used to calculate nodes' utilization and not the actual resource usage.
+func HighNodeUtilization(ctx context.Context, client clientset.Interface, strategy api.DeschedulerStrategy, nodes []*v1.Node, podEvictor *evictions.PodEvictor) {
+	if err := validateNodeUtilizationParams(strategy.Params); err != nil {
+		klog.ErrorS(err, "Invalid HighNodeUtilization parameters")
+		return
+	}
+	thresholdPriority, err := utils.GetPriorityFromStrategyParams(ctx, client, strategy.Params)
+	if err != nil {
+		klog.ErrorS(err, "Failed to get threshold priority from strategy's params")
+		return
+	}
+
+	thresholds := strategy.Params.NodeResourceUtilizationThresholds.Thresholds
+	targetThresholds := strategy.Params.NodeResourceUtilizationThresholds.TargetThresholds
```

Comment:
- `thresholds` is the low threshold
- `targetThresholds` is the high threshold

As discussed in https://github.com/kubernetes-sigs/descheduler/issues/196#issuecomment-814816526, please pin targetThresholds of all resources to 100%. Also, `validateNodeUtilizationParams` needs to make sure `strategy.Params.NodeResourceUtilizationThresholds.TargetThresholds` is never set until we allow it.